erigon-runtime/src/main.rs

```diff
@@ -0,0 +1,157 @@
+    "--http.corsdomain",
+    "*",
+    "--http.api",
+    "eth,debug,net,trace,web3,tg",
+];
+
+#[derive(Default, RuntimeDef)]
+pub struct ErigonRuntime {
+    seq: AtomicU64,
+    erigon_pid: Option<Child>,
+    rpcdaemon_pid: Option<Child>,
+}
+
+impl Runtime for ErigonRuntime {
+    const MODE: RuntimeMode = RuntimeMode::Server;
+
+    fn deploy<'a>(&mut self, _: &mut Context<Self>) -> OutputResponse<'a> {
+        async move {
+            Ok(serialize::json::json!(
+                {
+                    "startMode":"blocking",
+                    "valid":{"Ok":""},
+                    "vols":[]
+                }
+            ))
+        }
+        .boxed_local()
+    }
+
+    fn start<'a>(&mut self, ctx: &mut Context<Self>) -> OutputResponse<'a> {
+        let chain_id = String::from("goerli");
+        let current_exe_path = std::env::current_exe().unwrap();
+        let path = current_exe_path.parent().unwrap();
+
+        let erigon_pid = Command::new(&path.join(ERIGON_BIN))
+            .args(ERIGON_PARAMS)
+            .arg("--chain")
+            .arg(chain_id)
+            .spawn()
+            .expect("Erigon: Failed to spawn");
+
+        let rpcd_pid = Command::new(&path.join(RPCDAEMON_BIN))
+            .args(RPCDAEMON_PARAMS)
+            .spawn()
+            .expect("RPC Daemon: Failed to spawn");
+
+        self.erigon_pid = Some(erigon_pid);
+        self.rpcdaemon_pid = Some(rpcd_pid);
+
+        let emitter = ctx.emitter.clone().unwrap();
+        let stdout = format!(
+            "Erigon available at: http://{}:8545",
+            local_ipaddress::get().unwrap()
+        )
+        .as_bytes()
+        .to_vec();
+
+        async move {
+            emitter.command_stdout(0, stdout).await;
+            Ok("start".into())
+        }
+        .boxed_local()
+    }
+
+    fn stop<'a>(&mut self, _: &mut Context<Self>) -> EmptyResponse<'a> {
+        if let Some(erigon_pid) = &mut self.erigon_pid {
+            erigon_pid.kill().unwrap();
+        }
+        if let Some(rpcd_pid) = &mut self.rpcdaemon_pid {
+            rpcd_pid.kill().unwrap();
+        }
+
+        async move { Ok(()) }.boxed_local()
+    }
+
+    fn run_command<'a>(
+        &mut self,
+        command: RunProcess,
+        _mode: RuntimeMode,
+        ctx: &mut Context<Self>,
+    ) -> ProcessIdResponse<'a> {
+        let seq = self.seq.fetch_add(1, Relaxed);
+        let emitter = ctx.emitter.clone().unwrap();
+
+        let (tx, rx) = oneshot::channel();
+
+        tokio::task::spawn_local(async move {
+            // command execution started
+            emitter.command_started(seq).await;
+            // resolves the future returned by `run_command`
+            let _ = tx.send(seq);
+
+            let erigon_mock_data = serialize::json::json!(
+                {
+                    "status": "running",
+                    "url": format!("http://{}:8545", local_ipaddress::get().unwrap()),
+                    "secret": "THE SECRET AUTH"
```

Comment:
It should be empty until the authentication is implemented (because webserver is supposed to append this secret to the URL so nothing breaks if it is empty).